graphqlapi/src/main/java/io/stargate/graphql/schema/graphqlfirst/processor/UpdateModelBuilder.java

```diff
@@ -108,6 +110,9 @@ MutationModel build() throws SkipException {
       incrementModels = directives.getIncrementModels();
     }
 
+    // we are using GraphQLString because it will be coerced to BigInteger (Long)
```

Comment:
I'm not sure I follow. Why a string instead of something numeric?

Reply:
`BigInteger` is represented as `String` in GraphQL

Reply:
To be clear, we allow both `Int` and `String` (it's the user's choice, they define the type of the argument in their schema).

Large integer literals are subject to interoperability issues (the server receives truncated values). This only happens for values larger than (2^53 - 1). Timestamps are in microseconds, so they will reach that limit in the year 2255.

Most applications deal with contemporary timestamps, so they can safely choose `Int`. But if for any reason they were to send timestamps over the limit, then they must use `String`.

All this should be explained in the description of the `@cql_timestamp` directive.

Reply:
Actually, it should be `BigInt`, not `String`:
```java
    Optional<String> cqlTimestampArgumentName =
        findFieldNameWithDirective(
            CqlDirectives.TIMESTAMP, Arrays.asList(Scalars.GraphQLInt, CqlScalar.BIGINT.getGraphqlType()));
```
That way the coercion is already done automatically for you (instead of doing it manually in `TimestampParser`), and you'll get the value as a `Long` directly in the environment.

If you decide to support ISO strings per the other comment below, then you'll have `GraphQLString` as a third option.

Reply:
One more thing 😬 

I just realized that allowing `GraphQLInt` makes no sense. The highest value it can represent is `Integer.MAX_VALUE` microseconds, or about 35 minutes from the epoch.

So the only type we should allow are:
*  `CqlScalar.BIGINT.getGraphqlType()`
* *if we decide to support ISO strings*, `Scalars.GraphQLString`.

Reply:
Regarding using `CqlScalar.BIGINT` instead of `GraphQLString` to carry the value of `BigInt`, I am not sure how it suppose to work automatically. 
It is impossible to create such a GraphQL schema:
```
       + " updateWithWriteTimestamp(\n"
            + "    k: Int\n"
            + "    v: Int\n"
            + "    write_timestamp: BigInt @cql_timestamp\n"
            + "  ): Boolean\n"
```
, it throws:
```
       + " updateWithWriteTimestamp(\n"
            + "    k: Int\n"
            + "    v: Int\n"
            + "    write_timestamp: BigInt @cql_timestamp\n"
            + "  ): Boolean\n"
```
So the `write_timestamp` must be passed as `String`. Even when having this:
```
       findFieldNameWithDirective(
            CqlDirectives.TIMESTAMP,
            Arrays.asList(CqlScalar.BIGINT.getGraphqlType(), Scalars.GraphQLString));
```
The ` environment.getArgument(name);` for the `write_timestamp` returns a `String`  (not `Long`) so there is a need for manual coercion. Am I missing something?

Reply:
I think you meant to paste an error but you pasted the schema a second time.

I tried locally. After passing the right types to `findFieldNameWithDirective` there is indeed an issue: CQL scalars are only added to the schema if they are referenced, and that is currently done only for entity fields. We need to do it for regular arguments too.

I'm pushing a commit (b096de2b) since I have it already.